src/mcts/search.cc

```diff
@@ -108,9 +108,15 @@ void Search::SendUciInfo() REQUIRES(nodes_mutex_) {
   }
   common_info.hashfull =
       cache_->GetSize() * 1000LL / std::max(cache_->GetCapacity(), 1);
-  common_info.nps =
-      common_info.time ? (total_playouts_ * 1000 / common_info.time) : 0;
-  common_info.tb_hits = tb_hits_.load(std::memory_order_acquire);
+  if (nps_start_time_) {
+    const auto time_since_first_batch_ms =
+        std::chrono::duration_cast<std::chrono::milliseconds>(
+            std::chrono::steady_clock::now() - *nps_start_time_)
+            .count();
+    common_info.nps = total_playouts_ * 1000 / time_since_first_batch_ms;
+  } else {
+    common_info.nps = 0;
+  }
```

Comment:
Here, `time_since_first_batch_ms` may be zero - e.g. by doing `go nodes 100` twice.  Suggested patch:
```diff
diff --git a/src/mcts/search.cc b/src/mcts/search.cc
index 6772688..a233de0 100644
--- a/src/mcts/search.cc
+++ b/src/mcts/search.cc
@@ -113,9 +113,9 @@ void Search::SendUciInfo() REQUIRES(nodes_mutex_) {
         std::chrono::duration_cast<std::chrono::milliseconds>(
             std::chrono::steady_clock::now() - *nps_start_time_)
             .count();
-    common_info.nps = total_playouts_ * 1000 / time_since_first_batch_ms;
-  } else {
-    common_info.nps = 0;
+    if (time_since_first_batch_ms > 0) {
+      common_info.nps = total_playouts_ * 1000 / time_since_first_batch_ms;
+    }
   }
 
   int multipv = 0;
```
This way nps is not reported if unknown. The last else is also removed to do the same when `nps_start_time_` is still unset.

Reply:
Done.